common/ledger/blkstorage/blockfile_mgr.go

```diff
@@ -461,7 +512,12 @@ func (mgr *blockfileMgr) retrieveBlockByNumber(blockNum uint64) (*common.Block,
 	if blockNum == math.MaxUint64 {
 		blockNum = mgr.getBlockchainInfo().Height - 1
 	}
-
+	if blockNum < mgr.firstPossibleBlockNumberInBlockFiles() {
+		return nil, errors.Errorf(
+			"cannot serve block [%d]. The ledger is bootstrapped from a snapshot. First available block = [%d]",
+			blockNum, mgr.firstPossibleBlockNumberInBlockFiles(),
+		)
```

Comment:
Maybe we need to introduce an error type? While other filesystem errors might result in panic or stop processing of a chain, this shouldn't be right?

Reply:
This is called when serving a user query. Do you see a panic in the call path that leads to this? Otherwise as well, in general we need to rather remove the panics at the lower layers of the code and propagate errors to the top and panic decision should better be taken at the highest level.